File: src/main/java/com/wooteco/wiki/dto/DocumentRecentResponse.java
package com.wooteco.wiki.dto;

import com.wooteco.wiki.entity.Document;
import java.time.LocalDateTime;

public record DocumentRecentResponse(Long documentId, String title, LocalDateTime generateTime) {
    public static DocumentRecentResponse from(Document document) {
        return new DocumentRecentResponse(document.getDocumentId(), document.getTitle(), document.getGenerateTime());
    }
}

File: src/main/java/com/wooteco/wiki/dto/DocumentRecentResponse.kt
package com.wooteco.wiki.dto

import com.wooteco.wiki.domain.Document
import java.time.LocalDateTime
import java.util.*

data class DocumentRecentResponse(
    val documentId: Long?,
    val uuid: UUID,
    val title: String,
    val generateTime: LocalDateTime,
) {
    companion object {
        fun from(document: Document) = DocumentRecentResponse(
            documentId = document.documentId,
            uuid = document.uuid,
            title = document.title,
            generateTime = document.generateTime,
        )
    }
}
